### PR TITLE
Enhance integration tests w/ lineage and report metadata checks

### LINKED ISSUE #15582: Enhance lineage assertions in integration tests
### User Story
As a developer I want comprehensive tests so that I can catch any issues before they go to production.

### Description/Use Case
Many of the integration tests for `FHIRConverter`, `FHIRDestinationFilter`, `FHIRReceiverFilter`, and `FHIRTranslator` aren't checking the preceding lineage and/or the metadata on output `Report` objects. We should add these checks where it makes sense.

### Risks/Impacts/Considerations
This ticket originally only covered destination and receiver filters but has been expanded to cover all FHIR function integration tests.

### Dev Notes
These live in `FHIRConverterIntegrationTests.kt`, `FHIRDestinationFilterIntegrationTests.kt`, `FHIRReceiverFilterIntegrationTests.kt`, and `FHIRTranslatorIntegrationTests.kt`

This ticket got closed as a duplicate but contains helpful context for this ticket and all the AC there should be met here as well: https://github.com/CDCgov/prime-reportstream/issues/14349

### Acceptance Criteria
- `FhirDestinationFilterIntegrationTests`
  - [x] `Report` objects retrieved; relevant properties checked when report does not get routed anywhere
  - [x] `Report` object retrieved; contents and properties checked when report is routed
- `FhirReceiverFilterIntegrationTests`
  - [x] `Report` object retrieved; contents and properties checked when report does not get routed anywhere
- Verify lineages are adequately checked in `fetchChildReports`
  - [x] Double-check the lineage assertions for ItemLineage and ReportLineage
  - [x] Revisit usage (or lack of) in other integration tests (Convert and Translate) and ~~create tickets as needed~~ resolve

### PR DESCRIPTION
**Pull request by thetaurean**
This PR enhances the integration tests for the Convert, Destination Filter, Receiver Filter, and Translate steps in the UP. Previously we were not checking lineage and report metadata in all cases.

Test Steps:
1. Run test suites

## Changes
- Added lineage and report metadata assertions to `FHIRConverter`, `FHIRDestinationFilter`, `FHIRReceiverFilter`, and `FHIRTranslator`

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] ~~Added~~ Updated tests?

## Linked Issues
- Fixes #15582

### CODE REVIEW

**Review by adegolier — APPROVED** [review]
Tests passed when I ran them, LGTM. Can't want to see if this fixes the other issues.

File: prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiverFilter.kt [hunk 1]
```diff
@@ -357,6 +357,7 @@ class FHIRReceiverFilter(
                         ),
                         metadata = this.metadata,
                         topic = queueMessage.topic,
+                        destination = receiver,
```
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiverFilter.kt:360]: We are assuming this change will fix https://github.com/CDCgov/prime-reportstream/issues/15645? Can we please confirm that and, if so, update this PR description to link to fixing that ticket?

**Review by arnejduranovic — APPROVED** [review]

File: prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiverFilter.kt (hunk 1 above)
Comment (adegolier) [reply prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiverFilter.kt:360]: We don't know if that fixes it or not.

Comment (thetaurean) [reply prime-router/src/main/kotlin/fhirengine/engine/FHIRReceiverFilter.kt:360]: This did not fix either bug (though it is required to fix them). More details to come but I think we should merge this and continue fixing the bugs in a separate PR.

**Pull request merged**